### PR TITLE
add missing SOF_TIMESTAMPING flags

### PR DESCRIPTION
**Pull request by folkertdev**
These flags are useful for send (tx) timestamping in practice, and used by @tweedegolf's work on PTP and NTP

### CODE REVIEW

**Comment by folkertdev** [comment]
I must be missing something about the test setup here. Does anyone know why the CI build fails?

File: src/unix/linux_like/linux/mod.rs [hunk 1]
```diff
@@ -2673,6 +2673,15 @@ pub const SOF_TIMESTAMPING_RX_SOFTWARE: ::c_uint = 1 << 3;
...
 pub const SOF_TIMESTAMPING_RAW_HARDWARE: ::c_uint = 1 << 6;
+pub const SOF_TIMESTAMPING_OPT_ID: ::c_uint = 1 << 7;
+pub const SOF_TIMESTAMPING_TX_SCHED: ::c_uint = 1 << 8;
+pub const SOF_TIMESTAMPING_TX_ACK: ::c_uint = 1 << 9;
+pub const SOF_TIMESTAMPING_OPT_CMSG: ::c_uint = 1 << 10;
+pub const SOF_TIMESTAMPING_OPT_TSONLY: ::c_uint = 1 << 11;
+pub const SOF_TIMESTAMPING_OPT_STATS: ::c_uint = 1 << 12;
+pub const SOF_TIMESTAMPING_OPT_PKTINFO: ::c_uint = 1 << 13;
+pub const SOF_TIMESTAMPING_OPT_TX_SWHW: ::c_uint = 1 << 14;
+pub const SOF_TIMESTAMPING_BIND_PHC: ::c_uint = 1 << 15;
```
Comment (JohnTitor) [inline src/unix/linux_like/linux/mod.rs:2684]: `SOF_TIMESTAMPING_BIND_PHC` is newly added and our CI doesn't know it: https://github.com/torvalds/linux/commit/d463126e23f112629edb01594141ca437a92a108

We could skip the test for it.

**Comment by folkertdev** [comment]
@JohnTitor I removed the `BIND_PHC` flag from this PR. it is used in PTP rather than NTP and so it's likely it depends on kernel feature flags. I also don't need it specifically. So I'm happy with this PR as-is

**Comment by JohnTitor** [comment]
Thanks! @bors r+

**Pull request merged**